prov/gni/src/gnix_av.c

```diff
@@ -544,6 +544,17 @@ int gnix_av_open(struct fid_domain *domain, struct fi_av_attr *attr,
 	return ret;
 }
 
+int _gnix_av_addr_retrieve(struct fid_av *av, fi_addr_t fi_addr,
+			  fi_addr_t *real_addr)
+{
+	size_t size = sizeof(fi_addr_t);
+	int ret = FI_SUCCESS;
+
+	ret = gnix_av_lookup(av, fi_addr, real_addr, &size);
+	if (ret == -FI_ETOOSMALL)
```

Comment:
@jshimek and I had a discussion on this, and he has some thoughts on how to avoid this slightly non-standard way of dealing with the return code.  I'm fine with that being in a separate commit.